validatetoken.go

```diff
@@ -335,3 +338,27 @@ func (claims *Claims) isValidForRole(configRole string) bool {
 
 	return false
 }
+
+func LogHttp(logger *log.Logger, message string, logHeaders bool, headers []string, statusCode int, request *http.Request) {
+	var logPayload = make(map[string]string)
+
+	if logHeaders {
+		if headers != nil {
+			for _, header := range headers {
+				logPayload[header] = request.Header.Get(header)
+			}
+		} else {
+			for header, value := range request.Header {
+				logPayload[header] = strings.Join(value, ",")
+			}
+		}
+	}
+
+	logPayload["StatusCode"] = strconv.Itoa(statusCode)
+	logPayload["Url"] = request.URL.String()
+	logPayload["Method"] = request.Method
+	logPayload["Error"] = message
+
+	jsonStr, _ := json.Marshal(logPayload)
```

Comment:
It's good practice to handle the error here:
jsonStr, err := json.Marshal(logPayload)
    if err != nil {
        logger.Printf("Error marshaling log payload to JSON: %v\n", err)
        return
    }

Reply:
fixed now!